app/models/project/Project.scala

```diff
@@ -446,7 +447,18 @@ case class Project(override val id: Option[Int] = None,
     * @return Project home page
     */
   def homePage: Page = Defined {
-    val page = new Page(this.id.get, Page.HomeName, Page.Template(this.name, Page.HomeMessage), false, -1)
+    var body = Page.HomeMessage
```

Comment:
This should only do it if one is not manually provided, yes? If that is not the behaviour with this PR, it should be.

Reply:
I'm not quite sure what you're saying. This scrapes the README if a source URL is provided, and if it is from GitHub.

Reply:
I'm asking if this will overwrite an existing manually-created README.

Reply:
When a Project is created, it creates a Home Page that has `Page.HomeMessage` in it. Lines 448 - 451 retrieve the Page from database, passing the `page` variable as a default. This changes the default to the README it scraped from GitHub. It's basically replacing the text from `Page.HomeMessage` with the README.